pkg/client/client.go

```diff
@@ -105,54 +105,54 @@ type client struct {
 
 // Create implements client.Client
 func (c *client) Create(ctx context.Context, obj runtime.Object, opts ...CreateOptionFunc) error {
-	_, ok := obj.(*unstructured.Unstructured)
+	u, ok := obj.(*unstructured.Unstructured)
```

Comment:
I'm not entirely certain as to what you're trying to do here, but you haven't accomplished what was requested in the linked issue.  That's about using the separate `Unstructured` interface instead of the literal struct implementation `unstructured.Unstructured`

Reply:
I have totally misunderstood the ticket. Do you mean by having a custom struct definition for unstructured instead of using unstructured.Unstructured from apimachinery?

Reply:
there's an equivalent interface to `unstructured.Unstructured` that lives in APIMachinery.  It's something like `runtime.Unstructured', and, IIRC, it's what all the underlying methods actually take in APIMachinery

Reply:
I hope this is what you meant 😅. would it be a good idea use this interface in pkg/cache too ?